File: seed/csharp-model/trace/src/SeedTrace/V2/Problem/FileInfoV2.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using SeedTrace.Core;

namespace SeedTrace.V2;

public record FileInfoV2
{
    [JsonPropertyName("filename")]
    public required string Filename { get; set; }

    [JsonPropertyName("directory")]
    public required string Directory { get; set; }

    [JsonPropertyName("contents")]
    public required string Contents { get; set; }

    [JsonPropertyName("editable")]
    public required bool Editable { get; set; }

    /// <summary>
    /// Additional properties received from the response, if any.
    /// </summary>
    [JsonExtensionData]
    public IDictionary<string, JsonElement> AdditionalProperties { get; internal set; } =
        new Dictionary<string, JsonElement>();

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/file_info_v_2.rb

module Seed
    module Types
        class FileInfoV2 < Internal::Types::Model
            field :filename, String, optional: false, nullable: false
            field :directory, String, optional: false, nullable: false
            field :contents, String, optional: false, nullable: false
            field :editable, Internal::Types::Boolean, optional: false, nullable: false
        end
    end
end
